src/CMakeLists.txt

```diff
@@ -225,6 +225,72 @@ if (ENABLE_JIT_PROFILING)
     target_link_libraries(core PRIVATE "${VTUNE_LIBRARY}")
 endif()
 
+# --- RetroAchievements ---
+message(STATUS "Adding RetroAchievements integration sources")
+
+target_sources(core PRIVATE
+    RetroAchievements/RAClient.cpp
+    RetroAchievements/RAClient.h
+    RetroAchievements/RAFunctions.h
+)
+
+target_include_directories(core PRIVATE
+    "${CMAKE_CURRENT_SOURCE_DIR}/RetroAchievements"
+    "${CMAKE_CURRENT_SOURCE_DIR}" 
+)
+
+message(STATUS "RetroAchievements support enabled")
+
+# --- CURL & SSL Configuration ---
+target_compile_definitions(core PRIVATE HAVE_CURL CURL_STATICLIB)
+
+if(DEFINED ENV{MSYSTEM_PREFIX})
+    set(CURL_LDIR "$ENV{MSYSTEM_PREFIX}/lib")
+else()
+    set(CURL_LDIR "C:/msys64/ucrt64/lib")
+endif()
+
+set(NGTCP2_CRYPTO "${CURL_LDIR}/libngtcp2_crypto_ossl.a")
+
+target_link_libraries(core PRIVATE 
```

Comment:
Why is this list of libraries hardcoded?

Same issue as above, in addition the list of static libraries cURL depends on can change from OS to OS, version to version, distro to distro, build to build...